test/v3/createPool/reClamm/reClamm.integration.test.ts

```diff
@@ -0,0 +1,246 @@
+            testAddress,
+            [DAI.address, BAL.address],
+            PERMIT2[chainId],
+        );
+
+        createPoolInput = {
+            poolType,
+            name: 'ReClamm Bal Dai',
+            symbol: '50BAL-50DAI',
+            tokens: [
+                {
+                    address: BAL.address,
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+                {
+                    address: DAI.address,
+                    rateProvider: zeroAddress,
+                    tokenType: TokenType.STANDARD,
+                    paysYieldFees: false,
+                },
+            ],
+            swapFeePercentage: parseUnits('0.01', 18),
+            pauseManager: zeroAddress,
+            swapFeeManager: zeroAddress,
+            initialMinPrice: parseUnits('0.5', 18),
+            initialMaxPrice: parseUnits('8', 18),
+            initialTargetPrice: parseUnits('3', 18),
+            priceShiftDailyRate: parseUnits('1', 18),
+            centerednessMargin: parseUnits('0.2', 18),
+            chainId,
+            protocolVersion,
+        };
+
+        poolAddress = await doCreatePool({
+            client,
+            testAddress,
+            createPoolInput,
+        });
+
+        // Get pool state
+        const initPoolDataProvider = new InitPoolDataProvider(chainId, rpcUrl);
+        poolState = await initPoolDataProvider.getInitPoolData(
+            poolAddress,
+            poolType,
+            protocolVersion,
+        );
+
+        // Take a snapshot after pool creation
+        snapshot = await client.snapshot();
+    }, 120_000);
+
+    beforeEach(async () => {
+        // Revert to the snapshot before each test
+        await client.revert({
+            id: snapshot,
+        });
+        // Take a new snapshot for the next test
+        snapshot = await client.snapshot();
+    });
+
+    test('pool should be created', async () => {
+        expect(poolAddress).to.not.be.undefined;
+    });
+
+    test('pool should be registered with Vault', async () => {
+        const isPoolRegistered = await client.readContract({
+            address: balancerV3Contracts.Vault[chainId],
+            abi: vaultExtensionAbi_V3,
+            functionName: 'isPoolRegistered',
+            args: [poolAddress],
+        });
+        expect(isPoolRegistered).to.be.true;
+    });
+
+    test('pool should init with BAL as given token', async () => {
+        // user chooses an amount for one of the tokens
+        const givenAmountIn = {
+            address: BAL.address,
+            rawAmount: parseUnits('69', BAL.decimals),
+            decimals: BAL.decimals,
+        };
+
+        // helper calculates the amount for the other token
+        const amountsIn = await calculateReClammInitAmounts({
+            createPoolInput,
+            tokens: poolState.tokens,
+            givenAmountIn,
+        });
```

Comment:
I kinda like asking for fewer inputs for `calculateReClammInitAmounts` helper

but we could also be more explicit if team prefers. this is all that is required from `createPoolInput`

```
   const { initialMinPrice, initialMaxPrice, initialTargetPrice } = createPoolInput;
```

Reply:
Since you're already using an object as input, it's quite a low effort to adapt it to be more explicit, right?
If you want to leverage the already existing `createPoolInput`, all you'd have to do is to pass it spreaded (i.e. `...createPoolInput`.
But definitely not a blocker in case you'd like to keep as is 👍

Reply:
I prefer more explicit as its easier to follow/debug and takes less work to construct the input.